### PR TITLE
Test

### PR DESCRIPTION
**Pull request by hafiz1379**
Hello 😍🙋‍♂️🙋‍♂️
🔰 In this project I:
✅ add tests to the Math Magicians app using [Jest](https://jestjs.io/) and [React Testing Library](https://testing-library.com/docs/react-testing-library/intro).
✅ Set up React Testing Library.
          ⏺ Follow the instructions in the official [documentation](https://testing-library.com/docs/react-testing-library/intro).
✅ Write unit tests for the files operate.js and calculate.js using Jest.
✅ Create unit tests for all React components:
          ⏺ Use Jest snapshots to test the components.
          ⏺ Use React Testing Library to simulate user interaction.

### CODE REVIEW

File: src/logic/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import calculate from './calculate';
+
+describe('calculate', () => {
+  it('should return an object with total, next, and operation properties when passed an AC buttonName', () => {
+    const result = calculate({ total: '5', next: '3', operation: '+' }, 'AC');
+    expect(result).toEqual({ total: null, next: null, operation: null });
+  });
+});
```
Comment (julie-ify) [inline src/logic/calculate.test.js]: - [ ] Kindly add more tests to cover all scenarios in the calculate file.

File: src/logic/operate.test.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import operate from './operate';
+
+describe('operate', () => {
+  it('should return the sum of two numbers when passed "+" as the operation', () => {
+    const result = operate('5', '3', '+');
+    expect(result).toEqual('8');
+  });
+});
```
Comment (julie-ify) [inline src/logic/operate.test.js]: - [ ] Kindly add more tests to cover all scenarios in the operate file.

File: src/components/Calculator.test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import { prettyDOM } from '@testing-library/dom';
+import Calculator from './Calculator';
+
+describe('Calculator', () => {
+  test('calculator is rendering', () => {
+    const calculator = render(<Calculator />);
+    expect(prettyDOM(calculator.container)).toMatchSnapshot();
```
Comment (julie-ify) [inline src/components/Calculator.test.js:10]: - [ ] Kindly add snapshot tests for the quote component. All components should have snapshot tests.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @hafiz1379 and @stino-x,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by topeogunleye — APPROVED** [review]
### Hi @hafiz1379, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- Added more tests to cover all scenarios in the calculate file :heavy_check_mark: 
- Added snapshot tests for the quote component :heavy_check_mark: 
 
## [Optional] suggestions:
- For future pull requests, please notice that your branch name `test` does not follow the convention of being named after a feature or action. For example, you could name your branch `added-test`. This makes it easier to track the changes and the purpose of each branch.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**